FileConverter.cc

```diff
@@ -0,0 +1,94 @@
+#include "FileConverter.h"
+
+using namespace carta;
+
+FileConverter::FileConverter(std::string root_folder) : _root_folder(root_folder) {}
+
+void FileConverter::SaveFile(const std::string& in_file, casacore::ImageInterface<float>* image, const CARTA::SaveFile& save_file_msg,
+    CARTA::SaveFileAck& save_file_ack) {
+    int file_id(save_file_msg.file_id());
+    std::string output_filename(save_file_msg.output_file_name());
+    std::string directory(save_file_msg.output_file_directory());
+    CARTA::FileType output_file_type(save_file_msg.output_file_type());
+
+    // Set response message
+    save_file_ack.set_file_id(file_id);
+    bool success = false;
+    casacore::String message;
+
+    // Get the full resolved name of the output image
+    std::string temp_path = _root_folder + "/" + directory;
+    std::string abs_path = fs::absolute(temp_path);
+    output_filename = abs_path + "/" + output_filename;
+
+    if (output_filename == in_file) {
+        message = "The source file can not be overwritten!";
+        save_file_ack.set_success(success);
+        save_file_ack.set_message(message);
+        return;
+    }
+
+    // Check the writing permission
```

Comment:
specifically, we should use `fs`:

```
#ifdef _BOOST_FILESYSTEM_
#include <boost/filesystem.hpp>
namespace fs = boost::filesystem;
#else
#include <filesystem>
namespace fs = std::filesystem;
#endif
```
since the MacOS 10.14 C++17 implementation is not feature-complete.